bin/makePsfLibrary.py

```diff
@@ -0,0 +1,175 @@
+            if len(line) == 0:
+                break
+            # catalog lines with objects have x,y position at index 2,3.  Use that for 1st guess
+            attrs = line.split()
+            if attrs[0] != "object": continue
+            coord = afwCoord.Coord(afwGeom.Point2D(float(attrs[2]), float(attrs[3])))
+            pixels = exp.getWcs().skyToPixel(coord)
+            x = int(pixels.getX())
+            y = int(pixels.getY())
+            starBBox = afwGeom.Box2I(afwGeom.Point2I(x, y), afwGeom.Extent2I(1,1))
+            starBBox.grow(psfSize)
+        
+            # test to be sure that the star is completely contained
+            if exp.getBBox().contains(starBBox):
+                subImage = afwImage.ImageF(exp.getMaskedImage().getImage(), starBBox)
+                ds = afwDetection.FootprintSet(subImage, afwDetection.Threshold(1000))
+                sources = afwTable.SourceCatalog(schema)
+                record = sources.addNew()
+                record.setFootprint(ds.getFootprints()[0])
+                plugin.measure(record, exp)
+        
+                # shift the bounding box to the correct centroid
+                newCoord = exp.getWcs().pixelToSky(afwGeom.Point2D(record.get("_x"), record.get("_y")))
+                dx = int(round(record.get("_x"))) - x
+                dy = int(round(record.get("_y"))) - y
+                starBBox.shift(afwGeom.Extent2I(dx, dy))
+        
+                # if the exposure still contains the cutout, use it.
+                if exp.getBBox().contains(starBBox):
+                    subImage = afwImage.ImageF(exp.getMaskedImage().getImage(), starBBox)
+        
+                    # convert the image to 64 bits, and turn it into a Psf
+                    # then warp it using the xytransform from above
+                    data = subImage.getArray().astype(numpy.float64)
+                    kernel = afwMath.FixedKernel(afwImage.ImageD(data))
+                    psf = measAlg.KernelPsf(kernel)
+                    warpedPsf = measAlg.WarpedPsf(psf, xytransform, "lanczos3");
+                    if len(outputHdus) == 0:
+                        hdu = pyfits.hdu.image.PrimaryHDU()
+                    else:
+                        hdu = pyfits.hdu.image.ImageHDU()
+                    hdu.data = warpedPsf.computeImage().getArray()
+                    hdu.header.append("GS_SCALE")
+                    hdu.header.set("GS_SCALE", .2)
+                    hdu.header.append("PSF_NO")
+                    hdu.header.set("PSF_NO",psfIndex)
+                    
+                    outputHdus.append(hdu)
+                    fout.write("%s %d %d\n"%(os.path.basename(output_file), psfCount, psfIndex))
+                    psfCount = psfCount + 1
+            psfIndex = psfIndex + 1
+        cat.close()
+        del hdu
+        del exp
+    outputHdus.writeto(output_file, clobber=True)
+    print psfCount, " psf images were cutout from the original image", len(outputHdus)
```

Comment:
When I tried this, it prints:
```
178  psf images were cutout from the original image 178
```
I've not yet fully digested the code above, but from a quick glance it looks as though ``psfCount`` and ``len(outputHdus)`` are always going to be the same. Do you need them both? Do you need _to print_ them both?